packages/medusa/src/utils/get-product-pricing-with-pricing-module.ts

```diff
@@ -0,0 +1,124 @@
+import { Request } from "express"
+import { PriceSelectionContext } from "../interfaces"
+import { MoneyAmount, ProductVariant } from "../models"
+import { isDefined } from "medusa-core-utils"
+import { ProductVariantPricing } from "../types/pricing"
+
+export async function getProductPricingWithPricingModule(
+  req: Request,
+  variants: ProductVariant[],
+  context: PriceSelectionContext = {}
+) {
+  const remoteQuery = req.scope.resolve("remoteQuery")
+  const pricingModule = req.scope.resolve("pricingModuleService")
+
+  const variables = {
+    variant_id: variants.map((v) => v.id),
+  }
+
+  const priceSetQuery = `
+    query {
+      variants {
+        id
+        title
+        price {
+          price_set {
+            id
+          }
+        }
+      }
+    }
+   `
+
+  const res: VariantsRes[] = await remoteQuery(priceSetQuery, {
+    context: variables,
+  })
+
+  const variantIdToPriceSetIdsMap = new Map(
+    res.map((r) => {
+      if (!r.price) {
+        return [r.id, []]
+      }
+
+      const value = Array.isArray(r.price)
+        ? r.price.map((p) => p.price_set_id)
+        : [r.price.price_set_id]
+      return [r.id, value]
+    })
+  )
+
+  const priceSetIds = res
+    .map((r) => {
+      if (!r.price) {
+        return []
+      }
+      return Array.isArray(r.price)
+        ? r.price.map((p) => p.price_set_id)
+        : [r.price.price_set_id]
+    })
+    .flat()
+
+  const queryContext: PriceSelectionContext = removeNullish(context)
+
+  if (queryContext.currency_code) {
+    queryContext.currency_code = queryContext.currency_code.toUpperCase()
+  }
+
+  const prices = await pricingModule.calculatePrices(
+    { id: priceSetIds },
+    {
+      context: queryContext,
+    }
+  )
+
+  const priceSetMap = new Map<string, MoneyAmount>(prices.map((p) => [p.id, p]))
+
+  variants.map((v) => {
+    const priceSetIds = variantIdToPriceSetIdsMap.get(v.id)
+    const prices = priceSetIds?.map((id) => priceSetMap.get(id)).filter(Boolean)
+    if (prices?.length) {
+      const calculatedPrice = prices.reduce((acc: number, curr): number => {
+        return !curr || acc < curr.amount ? acc : curr.amount
```

Comment:
Comment: when as user you read this, you have to look all around to know what things are. Harder to debug, fix, read etc. Naming is so important